### PR TITLE
Storage load limit

### PR DESCRIPTION
**Pull request by matei-p**

### CODE REVIEW

File: arwen/contexts/storage.go [hunk 1]
```diff
@@ -148,42 +156,60 @@ func (context *storageContext) GetStorageFromAddress(address []byte, key []byte)
...
-	return value, usedCache
+	return value, usedCache, nil
+}
+
+func (context *storageContext) readFromBlockchain(address []byte, key []byte) ([]byte, error) {
+	value, _, err := context.blockChainHook.GetStorageData(address, key)
+	if err != nil {
+		enableEpochsHandler := context.host.EnableEpochsHandler()
+		if enableEpochsHandler.IsMaxBlockchainHookCountersFlagEnabled() {
+			return nil, err
```
Comment (sasurobert) [inline arwen/contexts/storage.go:202]: Not all errors are the same. We should only care about MaxCounters error. Other error should be ignored. Maybe ignoring them in blockchainhook level is better.

File: arwen/contexts/storage.go [hunk 2]
```diff
@@ -148,42 +156,60 @@ func (context *storageContext) GetStorageFromAddress(address []byte, key []byte)
...
 	}
 
-	return value, usedCache
+	return value, usedCache, nil
+}
+
+func (context *storageContext) readFromBlockchain(address []byte, key []byte) ([]byte, error) {
+	value, _, err := context.blockChainHook.GetStorageData(address, key)
+	if err != nil {
+		enableEpochsHandler := context.host.EnableEpochsHandler()
```
[the remark this answers is not in the document]
Comment (iulianpascalau) [reply arwen/contexts/storage.go:200]: will remove the flag

File: arwen/contexts/storage.go (hunk 1 above)
Comment (iulianpascalau) [reply arwen/contexts/storage.go:202]: right, will refactor as to ensure that only the counter errors passthrough

**Review by sasurobert — DISMISSED** [review]

File: config/gasCost.go [hunk 3]
```diff
@@ -4,6 +4,7 @@ import "github.com/ElrondNetwork/wasm-vm-v1_4/wasmer"
 
 type GasCost struct {
 	BaseOperationCost    BaseOperationCost
+	MaxPerTransaction    MaxPerTransaction
```
Comment (camilbancioiu) [inline config/gasCost.go:7]: Not necessary anymore, can be deleted.

Comment (iulianpascalau) [reply config/gasCost.go]: Fixed

File: arwen/contexts/storage_test.go [hunk 4]
```diff
@@ -20,17 +22,49 @@ var elrondReservedTestPrefix = []byte("RESERVED")
...
+		}
+		host := &contextmock.VMHostMock{
+			EnableEpochsHandlerField: enableEpochsHandler,
+		}
+		mockBlockchain := worldmock.NewMockWorld()
+
+		storageContextInstance, err := NewStorageContext(host, mockBlockchain, elrondReservedTestPrefix)
+		require.Nil(t, err)
+		require.False(t, check.IfNil(storageContextInstance))
+	})
```
Comment (camilbancioiu) [inline arwen/contexts/storage_test.go:67]: :+1:

File: arwen/contexts/storage_test.go [hunk 5]
```diff
@@ -79,35 +113,35 @@ func TestStorageContext_SetAddress(t *testing.T) {
 	}
 	bcHook := &contextmock.BlockchainHookStub{}
 
-	storageContext, _ := NewStorageContext(host, bcHook, elrondReservedTestPrefix)
+	storageContextInstance, _ := NewStorageContext(host, bcHook, elrondReservedTestPrefix)
```
Comment (camilbancioiu) [inline arwen/contexts/storage_test.go:116]: Optionally rename to `storageCtx`.

File: arwen/contexts/storage_test.go [hunk 6]
```diff
@@ -373,86 +407,150 @@ func TestStorageContext_StorageProtection(t *testing.T) {
...
-	mockOutput.OutputAccountMock = account
-	mockOutput.OutputAccountIsNew = false
+	t.Run("blockchain hook errors", func(t *testing.T) {
+		t.Parallel()
 
-	mockRuntime := &contextmock.RuntimeContextMock{}
-	mockMetering := &contextmock.MeteringContextMock{}
-	mockMetering.SetGasSchedule(config.MakeGasMapForTests())
-	mockMetering.BlockGasLimitMock = uint64(15000)
+		scAddress := []byte("account")
```
Comment (camilbancioiu) [inline arwen/contexts/storage_test.go:440]: `scAddress` appears common to both subtests, could be moved outside of them, within the test function.

File: arwen/contexts/storage_test.go (hunk 5 above)
Comment (iulianpascalau) [reply arwen/contexts/storage_test.go:116]: renamed

File: arwen/contexts/storage_test.go (hunk 6 above)
Comment (iulianpascalau) [reply arwen/contexts/storage_test.go:440]: done

**Review by sasurobert — APPROVED** [review]

**Review by camilbancioiu — APPROVED** [review]

**Pull request merged**